cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MetricsConsumerMetaTable.java

```diff
@@ -65,20 +63,27 @@ public MetricsConsumerMetaTable(MetricsTable metaTable) {
 
 
   public <T extends MetricsMetaKey>
-  void saveMetricsProcessorStats(Map<T, MetricsProcessorStats> messageIds) throws Exception {
+  void saveMetricsProcessorStats(Map<T, TopicProcessMeta> messageIds) throws Exception {
     SortedMap<byte[], SortedMap<byte[], Long>> timestampUpdates = new TreeMap<>(Bytes.BYTES_COMPARATOR);
-    for (Map.Entry<T, MetricsProcessorStats> entry : messageIds.entrySet()) {
-      SortedMap<byte[], Long> timeMap = new TreeMap<>(Bytes.BYTES_COMPARATOR);
-      MetricsProcessorStats metaInfo = entry.getValue();
+    SortedMap<byte[], SortedMap<byte[], byte[]>> messageIdUpdates = new TreeMap<>(Bytes.BYTES_COMPARATOR);
+
+    for (Map.Entry<T, TopicProcessMeta> entry : messageIds.entrySet()) {
+      TopicProcessMeta metaInfo = entry.getValue();
       if (metaInfo.getMessagesProcessed() > 0L) {
+        SortedMap<byte[], Long> timeMap = new TreeMap<>(Bytes.BYTES_COMPARATOR);
         timeMap.put(PROCESS_COUNT_TOTAL, metaInfo.getMessagesProcessed());
```

Comment:
didn't we say we want to remove processCountTotal and instead have the number of records processed in the last fetch? If you want to, you can keep the total count, but please add the last processed count.